File: Java/xProjectDemoJava/app/src/main/java/com/example/xprojectdemojava/ui/share/ShareViewModel.java
package com.example.xprojectdemojava.ui.share;
//share es cosecha
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class ShareViewModel extends ViewModel {

    private MutableLiveData<String> mText;

    public ShareViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("Cosechas");
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: xProjectDemo/app/src/main/java/com/example/xprojectdemo/ui/share/ShareViewModel.kt
package com.example.xprojectdemo.ui.share

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.xprojectdemo.ui.home.mApp

class ShareViewModel : ViewModel() {

    private val _text = MutableLiveData<String>().apply {
        value = "This is share Fragment"

        mApp.xGlobalsVar.image="popo5"
    }
    val text: LiveData<String> = _text
}